src/opamp.rs

```diff
@@ -96,6 +96,18 @@ pub enum NonInvertingGain {
     Gain64 = 5,
 }
 
+/// Internal output of DAC 3 channel 1.
+pub struct Dac3Out1;
+
+/// Internal output of DAC 3 channel 2.
+pub struct Dac3Out2;
+
+/// Internal output of DAC 4 channel 1.
+pub struct Dac4Out1;
+
+/// Internal output of DAC 4 channel 2.
+pub struct Dac4Out2;
+
```

Comment:
We have the channel types from the [dac module](https://github.com/davidstalnaker/stm32g4xx-hal/blob/opamp-dac/src/dac.rs#L73) if we want to use those. I believe that would force the user to pass in the DAC they actually initialized, reducing the risk of error.

Reply:
I like the idea but I don't think taking ownership of the DAC channel works in this case, for two reasons:

1. The same DAC output can be used as input for multiple opamps (dac3ch1 can go to opamps 1 and 6 simultaneously, verified that this works).
2. The user wouldn't be able to set the DAC's value (or other configuration) after setting up the opamp.  There's some cases where that's ok (if using DMA) but the easiest setup would become impossible.

Reply:
Passing reference would probably be enough. As long as the OPAMP does not keep the reference, the dac can still be modified and similar which I believe would solve both 1 and 2?

Reply:
Something similar is done for the [comparator](https://github.com/davidstalnaker/stm32g4xx-hal/blob/opamp-dac/src/comparator.rs#L289).